### PR TITLE
demo message functionality implemented

### PR DESCRIPTION
**Pull request by mhamidjamil**
- [x] The implementation fulfills the desired requirement.
  <details>
    <summary>Details</summary>
    The code does the intended while also ensuring that it does not cause side-effects or compromise the system performance and functionality.
  </details>
- [x] The code does not add unnecessary complexity.
  <details>
    <summary>Details</summary>
    The code is neither too simple and long (over-engineering) nor too short and complex for peers. With a complex code, devs are likely to introduce bugs when calling or modifying the code.
  </details>
- [x] The code is style guide compliant.
  <details>
    <summary>Details</summary>
    The style guide of project is followed e.g. linting, formatting, types, and other standards. Any style point differing from the guide should not be combined with other changes, as it makes it hard to see what is being changed.
  </details>
- [x] The code follows agreed design principles.
  <details>
    <summary>Details</summary>
    SOLID principles are used efficiently, making the code reusable and scalable.
  </details>
- [x] The code is readable.
  <details>
    <summary>Details</summary>
    The code is readable and comprehensive. Comments are clear, useful, and explain why instead of what.
  </details>
- [x] The code has appropriate tests.
  <details>
    <summary>Details</summary>
    Tests should be added, as appropriate, in the same change list as the code. Tests should be correct, sensible, and useful.
  </details>
- [ ] Documentation has been updated.
  <details>
    <summary>Details</summary>
    Devs should update associated documentation. If code is deleted or deprecated, then the documentation should also be deleted.
  </details>

### CODE REVIEW

File: v5/v5_with_library/v5_stable.ino [hunk 1]
```diff
@@ -617,49 +618,53 @@ String getResponse() {
...
+      // pending_test_message = false;
+      newMessageNumber = demo_message_index.toInt();
+      senderNumber = demo_message_number;
+      _message_ = demo_message_content;
+    } else {
+      newMessageNumber = getNewMessageNumber(response);
+      senderNumber = getMobileNumberOfMsg(String(newMessageNumber), true);
+      _message_ = (removeNewline(removeOk(readMessage(newMessageNumber))));
+    }
+    if (!isIn(senderNumber, "3374888420")) {
```
Comment (wasey-rao) [inline v5/v5_with_library/v5_stable.ino:636]: I suggest this string should also move to config or SPI memory.

Comment (wasey-rao) [reply v5/v5_with_library/v5_stable.ino:636]: But you can mark it as resolve as I'm already late in merging. But please move this string to config or SPI.

File: v5/v5_with_library/v5_stable.ino [hunk 2]
```diff
@@ -617,49 +618,53 @@ String getResponse() {
...
-                String(newMessageNumber) + "} message: > [ " +
-                _message_.substring(0, _message_.indexOf(" <not executed>")) +
-                " ] from: " + senderNumber + ". deleting it...");
-          else
+          if (isValidString(_message_)) {
+            sendSMS("<Company message received sms no. {" +
+                    String(newMessageNumber) + "} message: > [ " + _message_ +
+                    " ] from: " + senderNumber + ". deleting it...");
+            if (newPackageSubscribed(_message_))
+              updatePackageSubscribedDate();
```
Comment (wasey-rao) [inline v5/v5_with_library/v5_stable.ino:665]: I like this approach, it improve code complexity unreadable company messages will no more executed to test the new packages subscription.

File: v5/v5_with_library/v5_stable.ino [hunk 3]
```diff
@@ -733,6 +739,8 @@ bool deleteMessage(int index) {
   } else if (index < 0) {
     index *= -1;
   }
+  if (pending_test_message)
+    return true;
```
Comment (wasey-rao) [inline v5/v5_with_library/v5_stable.ino:743]: Please explain me this? it will cause the loop, fix it i think you should assign it false not true.

Comment (mhamidjamil) [reply v5/v5_with_library/v5_stable.ino:743]: This is not in getResponse function it's in delete message.

**Pull request merged**